### PR TITLE
add eslint and prettier and apply

### PR DESCRIPTION
**Pull request by ananzh**
### Description
* add prettier and apply prettier
* add eslint and its config; apply eslint
* modify package.json script
* turn off no-unused-vars and no-dupe-else-if. will fix them in separate PRs
* modify workflow to run yarn lint for restricter format check
* remove standard

### Issue Resolved:
https://github.com/opensearch-project/opensearch-js/issues/243

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by ananzh** [comment]
unit and integration:

```
yarn test:unit
yarn run v1.22.5
$ tap test/unit/{*,**/*}.test.js
 RUNS  test/unit/api.test.js 217ms
 RUNS  test/unit/api.test.js 265ms
 SKIP  test/unit/esm/index.test.js
 ~ Skip because Node version < 12.17.0

 RUNS  test/unit/api.test.js 365ms
 SKIP  test/unit/esm/index.test.js 1 skip of 1 3.425ms
 ~ Skip because Node version < 12.17.0

 PASS  test/unit/selectors.test.js 8 OK 9.563ms
 PASS  test/unit/cloud-connection-pool.test.js 4 OK 14.875ms
 PASS  test/unit/helpers/search.test.js 8 OK 25.07ms
 PASS  test/unit/events.test.js 19 OK 33.283ms
 PASS  test/unit/errors.test.js 36 OK 29.342ms
 PASS  test/unit/api.test.js 25 OK 73.62ms
 PASS  test/unit/serializer.test.js 19 OK 33.465ms
 PASS  test/unit/base-connection-pool.test.js 44 OK 88.767ms
 PASS  test/unit/connection-pool.test.js 89 OK 168.016ms
 PASS  test/unit/child.test.js 36 OK 151.196ms
test/unit/helpers/bulk.test.js 2> (node:358889) DeprecationWarning: strictEqual() is deprecated, use equal() instead
test/unit/helpers/bulk.test.js 2> (node:358889) DeprecationWarning: deepEqual() is deprecated, use same() instead
 PASS  test/unit/helpers/scroll.test.js 58 OK 189.959ms
 PASS  test/unit/helpers/bulk.test.js 291 OK 244.887ms
 PASS  test/unit/helpers/msearch.test.js 49 OK 567.967ms
 PASS  test/unit/connection.test.js 73 OK 3s
 PASS  test/unit/transport.test.js 202 OK 4s
 PASS  test/unit/client.test.js 116 OK 11s

                         
  🌈 SUMMARY RESULTS 🌈  
                         

 SKIP  test/unit/esm/index.test.js 1 skip of 1 3.425ms
 ~ Skip because Node version < 12.17.0

Suites:   17 passed, 17 of 17 completed
Asserts:  1077 passed, 1 skip, of 1078
Time:     12s
Done in 12.34s.


yarn test:integration:helpers
yarn run v1.22.5
$ tap test/integration/helpers/*.test.js
 PASS  test/integration/helpers/search.test.js 11 OK 2s
 PASS  test/integration/helpers/msearch.test.js 27 OK 6s
 PASS  test/integration/helpers/scroll.test.js 215 OK 6s
 PASS  test/integration/helpers/bulk.test.js 25 OK 8s

                         
  🌈 SUMMARY RESULTS 🌈  
                         

Suites:   4 passed, 4 of 4 completed
Asserts:  278 passed, of 278
Time:     8s
Done in 8.71s.
```

**Comment by ananzh** [comment]
sanity test:
```
node data.js
Creating index:
{ acknowledged: true, shards_acknowledged: true, index: 'books' }
Cluster health:
{ cluster_name: 'runTask',
  status: 'yellow',
  timed_out: false,
  number_of_nodes: 1,
  number_of_data_nodes: 1,
  discovered_master: true,
  discovered_cluster_manager: true,
  active_primary_shards: 4,
  active_shards: 4,
  relocating_shards: 0,
  initializing_shards: 0,
  unassigned_shards: 12,
  delayed_unassigned_shards: 0,
  number_of_pending_tasks: 0,
  number_of_in_flight_fetch: 0,
  task_max_waiting_in_queue_millis: 0,
  active_shards_percent_as_number: 25 }
Cluster getSettings use master_timeout:
{ persistent: {}, transient: {} }
Cluster getSettings use cluster_manager_timeout:
{ persistent: {}, transient: {} }
Adding document:
{ _index: 'books',
  _id: '1',
  _version: 1,
  result: 'created',
  forced_refresh: true,
  _shards: { total: 4, successful: 1, failed: 0 },
  _seq_no: 0,
  _primary_term: 1 }
Search results:
{ total: { value: 1, relation: 'eq' },
  max_score: 0.5753642,
  hits:
   [ { _index: 'books',
       _id: '1',
       _score: 0.5753642,
       _source: [Object] } ] }
Deleting document:
{ _index: 'books',
  _id: '1',
  _version: 2,
  result: 'deleted',
  _shards: { total: 4, successful: 1, failed: 0 },
  _seq_no: 1,
  _primary_term: 1 }
Deleting index:
{ acknowledged: true }
```

File: .github/workflows/nodejs.yml [hunk 1]
```diff
@@ -24,9 +24,14 @@ jobs:
...
         npm install
 
+    - name: Setup Yarn
+      run: |
+        npm uninstall -g yarn
+        npm i -g yarn@1.22.10    
+
     - name: Lint
       run: |
-        npm run lint
```
Comment (kavilla) [inline .github/workflows/nodejs.yml:29]: How come? If yarn isn't a dev dependency?

Comment (ananzh) [reply .github/workflows/nodejs.yml:29]: I don't quite understand the question. To me, `npm run lint ` will run standard lint. But I want to setup some rules and for future management. So I switch to use `yarn lint`.

Comment (ananzh) [reply .github/workflows/nodejs.yml:29]: I will add yarn as devDependencies.

File: package.json [hunk 2]
```diff
@@ -33,10 +33,10 @@
     "test:coverage-90": "tap test/{unit,acceptance}/{*,**/*}.test.js --coverage --branches=90 --functions=90 --lines=90 --statements=90 --nyc-arg=\"--exclude=api\"",
     "test:coverage-report": "tap test/{unit,acceptance}/{*,**/*}.test.js --coverage --branches=90 --functions=90 --lines=90 --statements=90 --nyc-arg=\"--exclude=api\" && nyc report --reporter=text-lcov > coverage.lcov",
     "test:coverage-ui": "tap test/{unit,acceptance}/{*,**/*}.test.js --coverage --coverage-report=html --nyc-arg=\"--exclude=api\"",
-    "lint": "standard",
-    "lint:fix": "standard --fix",
+    "lint": "eslint .",
+    "lint:fix": "eslint . --fix",
     "license-checker": "license-checker --production --onlyAllow='MIT;Apache-2.0;Apache1.1;ISC;BSD-3-Clause;BSD-2-Clause'",
-    "build-esm": "npx gen-esm-wrapper . index.mjs && standard --fix index.mjs"
+    "build-esm": "npx gen-esm-wrapper . index.mjs && eslint --fix index.mjs"
```
Comment (tmarkley) [inline package.json:39]: Are we missing the `.` here: `eslint . --fix`?

File: package.json [hunk 3]
```diff
@@ -46,31 +46,36 @@
...
     "simple-statistics": "^7.7.0",
     "split2": "^3.2.2",
-    "standard": "^16.0.3",
     "stoppable": "^1.1.0",
     "tap": "^15.0.9",
     "tsd": "^0.15.1",
     "workq": "^3.0.0",
-    "xmlbuilder2": "^2.4.1"
+    "xmlbuilder2": "^2.4.1",
+    "yarn": "^1.22.18"
```
Comment (tmarkley) [inline package.json:78]: Shouldn't this go under the `engines` property?

File: .github/workflows/nodejs.yml [hunk 4]
```diff
@@ -13,29 +13,34 @@ jobs:
...
+          npm uninstall -g yarn
+          npm i -g yarn@1.22.18    
+    
+      - name: Lint
+        run: |
+          yarn lint
+
+      - name: Unit test
+        run: |
+          npm run test:unit
```
Comment (tmarkley) [inline .github/workflows/nodejs.yml:38]: Why are we only using `yarn` for linting and nothing else in this workflow?

File: package.json (hunk 3 above)
Comment (ananzh) [reply package.json:78]: right now we only use yarn for lint. so I added it to devDependency.

File: .github/workflows/nodejs.yml (hunk 4 above)
Comment (ananzh) [reply .github/workflows/nodejs.yml:38]: yes. it is currently the only used place

File: package.json (hunk 2 above)
Comment (ananzh) [reply package.json:39]: I think it only fix one file index.mjs so we don't need '.'
`eslint .` is to fix everything besides what are in `.eslintignore`, right?

Comment (tmarkley) [reply package.json:39]: Yes you are correct, I completely glossed over that.

File: package.json (hunk 3 above)
Comment (tmarkley) [reply package.json:78]: We have a `yarn.lock` file though so we can't only be using it for this?

Comment (ananzh) [reply package.json]: will do

**Review by VachaShah — APPROVED** [review]

**Pull request merged**